ipc_common.c

```diff
@@ -0,0 +1,68 @@
+#include "ipc_common.h"
+
+void options(ipc *info, int argc, char *argv[]){
```

Comment:
Structs can be returned by value from functions, it's minor since you only use this function once but I'd suggest changing the function signature as follows:
```suggestion
ipc options(int argc, char *argv[]){
```
Then creating the struct inside the function ```ipc info = (ipc) {.file_name=NULL, .method_name=NULL, .fflag=0, .qflag=0, .pflag=0, .sflag=0}```

And simply returning info

Reply:
I already had the IPC struct in a header file and I don't think it is necessary to make one more inside function.

Reply:
I'm not sure I understand, my suggestion was to allocate the "ipc" struct on the stack by returning an "ipc" value from options instead of allocating it on the heap through malloc and then modifying it via a pointer in "options"

Reply:
But in any case it was just a style suggestion, the only thing blocking the PR for me is the bug in #4